profiler/src/ProfilerEngine/Datadog.Profiler.Native/ProfileExporter.cpp

```diff
@@ -409,7 +409,10 @@ std::vector<UpscalingInfo> ProfileExporter::GetUpscalingInfos()
 
     for (auto& provider : _upscaledProviders)
     {
-        samplingInfos.push_back(provider->GetInfo());
+        for (auto& upscalingInfo : provider->GetInfos())
+        {
+            samplingInfos.push_back(upscalingInfo);
+        }
```

Comment:
you can use `std::list<T>::splice` instead to avoid copying object one by one.
```
samplingInfos.splice(sampligInfos.end(), provider->GetInfos());
```
this should work

Reply:
samplingInfos is a vector today and only a few elements will be copied + once per minute
Maybe check the impact in the next round of profiling measures
No need to change to a list<> everywhere